tests/integ/test_udf.py

```diff
@@ -1497,9 +1529,18 @@ def test_add_packages_negative(session, caplog):
         session.add_packages("python-dateutil****")
     assert "InvalidRequirement" in str(ex_info)
 
-    with pytest.raises(ValueError) as ex_info:
-        session.add_packages("dateutil")
-    assert "Cannot add package dateutil" in str(ex_info)
+    with patch.object(session, "_is_anaconda_terms_acknowledged", lambda: True):
+        with pytest.raises(RuntimeError) as ex_info:
+            session.add_packages("dateutil")
+
+        # dateutil is not a valid name, the library name is python-dateutil
+        assert "Pip failed with return code 1" in str(ex_info)
+
+    with patch.object(session, "_is_anaconda_terms_acknowledged", lambda: False):
```

Comment:
are the tests failing without the patch?

Reply:
Without the patch, anaconda terms are not acknowledged and RuntimeError will be raised, as expected in this test case. The patch helps ensure the test is not flaky (i.e. if the acknowledgement changes in the future, this test will remain robust.